websockets/README.md

```diff
@@ -0,0 +1,166 @@
+# Websockets in Apigee
+
+This sample shows how to deploy a sample websockets echo server in [Cloud Run](https://cloud.google.com/run) and how to use Apigee to expose that service to developers securely. For more information on how to use websockets in Apigee, please refer to the official [documentation](https://cloud.google.com/apigee/docs/api-platform/develop/websocket-config).
+
+## About websockets
+
+There are many situations where web interactions need to happen in real time, such as gaming, communications, financial transactions, and other high-throughput activities.
+
+WebSocket is a protocol that provides a full-duplex communications channel between a web client and web server over a single TCP connection. The WebSocket protocol uses the HTTP protocol to establish the connection between the client and server. Once established, the client and server then use the WebSocket protocol to send and receive data.
+
+The WebSockets [spec](https://websockets.spec.whatwg.org/) and protocol is maintained by the W3C.
+
+## How it works
+
+In Apigee and Apigee hybrid, environment groups provide routing to environments and define the hostnames on which API proxies are exposed. Environment groups support both the HTTP and WS protocols natively. You do not have to create a special environment group or any special configuration to use WebSockets. Rather, it is up to the client to request a protocol upgrade from HTTP to WS by including the Upgrade request header. An upgrade request made to an API proxy endpoint returns a 101 Switching Protocols response. Further requests and responses are made bidirectionally on the open WebSockets connection, until it is closed.
+
+### Policy support 
+With a WebSockets connection, you can only use the [Verify API Key](https://cloud.google.com/apigee/docs/api-platform/reference/policies/verify-api-key-policy) and [OAuthV2](https://cloud.google.com/apigee/docs/api-platform/reference/policies/oauthv2-policy) policies in your API proxy. All other policies are ignored.
+
+### Revoking the connection 
+
+The WebSockets connection is closed when:
+
+* The proxy endpoint receives a request without an API key or OAuth token.
+* The proxy endpoint receives a request with an expired or invalid API key or OAuth token.
+* The WebSockets connection times out.
+
+## Create a websockets backend in Cloud Run
```

Comment:
We should either move this to the setup section, or list this must be completed first under the prerequisites

Reply:
Also, is there a reason why we're having them manually do this, as opposed to automating it in the deploy script?

Reply:
Probably we should go back and update the other sample to make it consistent